src/js/components/conversation.jsx

```diff
@@ -62,7 +62,32 @@ export class ConversationComponent extends Component {
     }
 
     render() {
```

Comment:
Does this two-pass processing of the messages cause any slowdown in rendering speed? Should the messages be pre-processed on change, instead of re-processed on each render?